### PR TITLE
refactor: use SessionState directly

### PR DESCRIPTION
**Pull request by tustvold**
This shows how to use non-deprecated APIs to plan SQL queries without introducing potential state inconsistency

### CODE REVIEW

File: iox_query/src/exec/context.rs [hunk 1]
```diff
@@ -226,11 +230,12 @@ impl IOxSessionConfig {
 
         let inner = SessionContext::with_state(state);
 
+        // TODO: Clean up information schema handling on SessionContext
```
Comment (tustvold) [inline iox_query/src/exec/context.rs:233]: This was already on my list, the way that we shim the catalog providers is just a little strange

Comment (tustvold) [reply iox_query/src/exec/context.rs:233]: PR - https://github.com/apache/arrow-datafusion/pull/4740

File: iox_query/src/exec/context.rs [hunk 2]
```diff
@@ -308,13 +315,17 @@ impl IOxSessionContext {
         let ctx = self.child_ctx("prepare_sql");
         debug!(text=%sql, "planning SQL query");
 
-        // NOTE can not use ctx.inner.sql here as it also interprets DDL
-        // TODO: file ticket in DataFusion and fix
-        #[allow(deprecated)]
-        let logical_plan = ctx.inner.create_logical_plan(sql)?;
+        let mut statements = DFParser::parse_sql(sql)?;
```
Comment (alamb) [inline iox_query/src/exec/context.rs:318]: This API is reasonable to me. I think the `DFParser` is not super well documented at the moment, but I will go rectify that now 🏃

File: iox_query/src/exec/context.rs [hunk 3]
```diff
@@ -308,13 +315,17 @@ impl IOxSessionContext {
...
+        if statements.len() != 1 {
+            return Err(DataFusionError::NotImplemented(
+                "The context currently only supports a single SQL statement".to_string(),
```
Comment (alamb) [inline iox_query/src/exec/context.rs:321]:
```suggestion
                "IOx currently only supports a single SQL statement".to_string(),
```

**Review by alamb — APPROVED** [review]
makes sense to me 👍

**Comment by tustvold** [comment]
I intend to keep this open for a bit, as the API is still going through flux (these methods are going to move onto DataFrame), but I wanted to get something up to show the vague idea

**Comment by alamb** [comment]
I think the idea is good 👍

**Pull request closed without merging**